lib/media/segment_index.js

```diff
@@ -147,7 +147,9 @@ shaka.media.SegmentIndex.prototype.merge = function(references) {
             i == this.references_.length - 1 &&
             j == references.length - 1,
             'This should be an update of the last segment in a period');
-        newReferences.push(r2);
+        var r = new shaka.media.SegmentReference(r1.position,
+            r2.startTime, r2.endTime, r2.getUris, r2.startByte, r2.endByte);
+        newReferences.push(r);
```

Comment:
@TheModMaker, I don't think I understand you.  If we're processing both a new and old ref, and replacing one with the other, why wouldn't we increment the indices on both arrays?